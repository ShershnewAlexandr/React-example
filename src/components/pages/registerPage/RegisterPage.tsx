import React from 'react';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import RegisterFormContainer from '../../../containers/register/RegisterFormContainer';

function RegisterPage() {
  const { t } = useTranslation();
  return (
    <Grid container alignItems="center" justify="center">
      <Grid
        item
        className="editCardWrapper-container"
        container
        direction="column"
        justify="space-around"
        alignItems="center"
        spacing={2}
      >
        <Grid item>
          <span className="editCardWrapper-title">
            {t('register.registration')}
          </span>
        </Grid>
        <Grid item>
          <RegisterFormContainer />
        </Grid>
      </Grid>
    </Grid>
  );
}

export default RegisterPage;
