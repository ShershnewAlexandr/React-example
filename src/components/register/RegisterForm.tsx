import React from 'react';
import { Grid } from '@material-ui/core';
import { Field } from 'redux-form/immutable';
import { InjectedFormProps } from 'redux-form';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { routes } from '../../utils/constants';
import TextInput from '../inputs/TextInput';

function RegisterForm(props: InjectedFormProps) {
  const { handleSubmit, submitting } = props;
  const { t } = useTranslation();

  return (
    <form onSubmit={handleSubmit}>
      <Grid container direction="row" spacing={4}>
        <Grid item xs={12}>
          <Field
            name="email"
            component={TextInput}
            label={t('register.email')}
            disabled={submitting}
          />
        </Grid>
        <Grid item xs={12}>
          <Field
            name="password"
            component={TextInput}
            label={t('register.password')}
            disabled={submitting}
            isPassword={true}
          />
        </Grid>
        <Grid item xs={12}>
          <Field
            name="name"
            component={TextInput}
            label={t('register.name')}
            disabled={submitting}
          />
        </Grid>
        <Grid item container justify="center" spacing={2}>
          <Grid item>
            {!submitting && (
              <button
                type="submit"
                data-testid="registerButton"
                className="editCardWrapper-CancelButton"
              >
                {t('register.register')}
              </button>
            )}
          </Grid>
          <Grid item>
            {!submitting && (
              <Link to={routes.SIGNIN}>
                <button
                  type="submit"
                  data-testid="cancelButton"
                  className="editCardWrapper-CancelButton"
                >
                  {t('register.cancel')}
                </button>
              </Link>
            )}
          </Grid>
        </Grid>
      </Grid>
    </form>
  );
}

export default RegisterForm;
