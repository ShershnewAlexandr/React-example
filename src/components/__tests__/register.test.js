import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import RegisterPage from '../pages/registerPage/RegisterPage';
import RegisterForm from '../register/RegisterForm';

jest.mock('../../store/index');
jest.mock('i18next', () => ({
  use: function() {
    return this;
  },
  init: jest.fn(),
}));
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('registerPage', () => {
  const registerPage = shallow(<RegisterPage />);

  it('snapshot', () => {
    expect(toJson(registerPage)).toMatchSnapshot();
  });
});

describe('registerForm', () => {
  const props = {
    handleSubmit: jest.fn(),
  };

  const registerForm = shallow(<RegisterForm {...props} />);

  it('snapshot', () => {
    expect(toJson(registerForm)).toMatchSnapshot();
  });

  it('should catch submit', () => {
    registerForm.find('form').simulate('submit');
    expect(props.handleSubmit.mock.calls.length).toBe(1);
  });
});
