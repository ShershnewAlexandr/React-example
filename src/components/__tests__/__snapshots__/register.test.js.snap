// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerForm snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="row"
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Field
        component={[Function]}
        label="register.email"
        name="email"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Field
        component={[Function]}
        isPassword={true}
        label="register.password"
        name="password"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Field
        component={[Function]}
        label="register.name"
        name="name"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      justify="center"
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <button
          className="editCardWrapper-CancelButton"
          data-testid="registerButton"
          type="submit"
        >
          register.register
        </button>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Link
          to="/signin"
        >
          <button
            className="editCardWrapper-CancelButton"
            data-testid="cancelButton"
            type="submit"
          >
            register.cancel
          </button>
        </Link>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;

exports[`registerPage snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="editCardWrapper-container"
    container={true}
    direction="column"
    item={true}
    justify="space-around"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <span
        className="editCardWrapper-title"
      >
        register.registration
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <ReduxForm
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="register"
        getFormState={[Function]}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        validate={[Function]}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
